libraries/chain/apply_context.cpp

```diff
@@ -352,7 +355,8 @@ void apply_context::reset_console() {
 
 void apply_context::checktime(uint32_t instruction_count) {
    cpu_usage += instruction_count;
-   trx_context.add_cpu_usage_and_check_time( instruction_count );
+   EOS_ASSERT( BOOST_LIKELY(cpu_usage <= cpu_usage_limit), tx_cpu_usage_exceeded, "action cpu usage exceeded" );
```

Comment:
I would argue that BOOST_LIKELY (or the existing LIKELY of fc be part of EOS_ASSERT.